### PR TITLE
Remove "Git primer" section, fixes #74

### LINKED ISSUE #74: Git primer section
Probably controversial, but I think the "Git primer" section should vanish ([here](https://root-project.github.io/web/for_developers/git_primer/).

The website is about ROOT and not an introduction to git. Probably we can link to resources which we like, but such sections bloat the content of the website and hide the actual content related to ROOT.

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

File: contribute/index.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ Presently, there are two ways in which you can contribute:
 
     - Make sure you follow the [ROOT coding conventions]({{'/contribute/coding_conventions' | relative_url }}) in your code
     - Make sure you provide a set of tests for your feature/bug fix
+    - Make sure your git commits can be added to ROOT's master branch with a "rebase and merge", i.e. by fast-forwarding the master branch: ROOT tries to avoid merge commits as much as possible
```
Comment (pcanal) [inline contribute/index.md:34]: Reader might wonder 'why' ...

Comment (eguiraud) [reply contribute/index.md:34]: any suggestion to improve the text?

Comment (pcanal) [reply contribute/index.md:34]: (part of) the text in NoSY (No switch-yard) was intended as that explanation (maybe it failed :) ).

Comment (eguiraud) [reply contribute/index.md]: I inserted the line of the NoSY page that explained why we don't like merge commits here.

Comment (pcanal) [reply contribute/index.md]: thanks.

**Pull request merged**